tests/test_unit/test_auth/auth_fixtures/keycloak_fixture.py

```diff
@@ -81,8 +81,8 @@ def _create_session_cookie(user, expire_in_msec=1000) -> str:
 
 @pytest.fixture
 def mock_keycloak_well_known(settings):
-    server_url = settings.auth.server_url
-    realm_name = settings.auth.client_id
+    server_url = settings.auth.dict()["keycloak"]["server_url"]
```

Comment:
because when ServerAppSettings were initialized in runtime, it received SYNCMASTER__AUTH__KEYCLOAK__SERVER_URL=http://keycloak:8080 and added to settings as extra field, but at this time ServerAppSettings knows that there is nested auth_attribute where it can assign all attributes related to auth, but it knows nothing about keycloak settings and cannot assign keycloak object as settings attributes

Reply:
![Uploading image.png…]()